[BUG]: playwright-ci-smoke workflow crashes on startup — gateway never binds

### 🐞 Bug Summary

The `playwright-ci-smoke` CI job fails on every run. The gateway process exits immediately during startup and never binds to the port. All Playwright smoke tests time out waiting for `/health`.

---

### 🧩 Affected Component

- [x] CLI, Makefiles, or shell scripts

---

### 🔁 Steps to Reproduce

1. Open any PR that touches `mcpgateway/**`, `tests/playwright/**`, `pyproject.toml`, or `.github/workflows/playwright.yml`
2. Observe `playwright-ci-smoke` job
3. Gateway process exits at startup; health check loop exhausts 120 retries; job fails

---

### 🤔 Expected Behavior

Gateway starts, binds to port 4444, responds to `/health`, Playwright tests run.

---

### 📓 Logs / Error Output

```
Traceback (most recent call last):
  File ".venv/lib/python3.12/site-packages/gunicorn/app/base.py", line 110, in get_config_from_filename
    spec.loader.exec_module(mod)
  File "gunicorn.config.py", line 34, in <module>
    bind = f"{settings.host}:{settings.port}"
  File "mcpgateway/config.py", line 3552, in __getattr__
    return getattr(get_settings(), key)
  File "mcpgateway/config.py", line 3495, in get_settings
    cfg = Settings(**kwargs)
  File "mcpgateway/config.py", line 1469, in validate_security_combinations
    raise SecurityConfigurationError(
mcpgateway.config.SecurityConfigurationError: jwt_secret_key: Value is an unset placeholder (__REPLACE_ME__).
Run 'python -m mcpgateway.scripts.init_secrets' to generate strong values.
make: *** [Makefile:388: serve] Error 1
Gateway exited during startup.
```

---

### 🧠 Environment Info

| Key | Value |
|-----|-------|
| Version or commit | `main` |
| Runtime | Python 3.12, Gunicorn |
| Platform / OS | ubuntu-24.04 (GitHub Actions runner) |
| Container | none |

---

### 🧩 Root Cause

The workflow step copies `.env.example` to `.env` before launching the gateway:

```bash
cp .env.example .env
make serve > /tmp/mcpgateway-ci.log 2>&1 &
```

`.env.example` ships with placeholder values:

```
JWT_SECRET_KEY=__REPLACE_ME__run_init-secrets_before_starting
AUTH_ENCRYPTION_SECRET=__REPLACE_ME__run_init-secrets_before_starting
```

`validate_security_combinations()` in `mcpgateway/config.py` hard-fails on any value that starts with `__REPLACE_ME__` — by design, to prevent accidental deployment of uninitialized configs. No env vars in the workflow override these before `make serve` runs, so gunicorn crashes on load and the gateway process exits before binding.

**Fix:** Add `JWT_SECRET_KEY` and `AUTH_ENCRYPTION_SECRET` to the job-level `env:` block in `.github/workflows/playwright.yml`. pydantic-settings environment variables take precedence over `.env` file values, so the placeholder is never evaluated during the CI run.

--- a/.github/workflows/playwright.yml
+++ b/.github/workflows/playwright.yml
@@ -38,6 +38,9 @@ jobs:
       # Disable forced password-change on bootstrap so Playwright tests can
       # access /admin/ without being redirected to /admin/change-password-required
       ADMIN_REQUIRE_PASSWORD_CHANGE_ON_BOOTSTRAP: "false"
+      # Override .env.example placeholders — env vars take precedence over .env in pydantic-settings
+      JWT_SECRET_KEY: "ci-playwright-smoke-jwt-secret-not-for-production-use"
+      AUTH_ENCRYPTION_SECRET: "ci-playwright-smoke-auth-enc-secret-not-for-production"
 
     steps:
       - name: ⬇️ Checkout source

--- a/.github/workflows/playwright.yml
+++ b/.github/workflows/playwright.yml
@@ -39,8 +39,8 @@ jobs:
       # access /admin/ without being redirected to /admin/change-password-required
       ADMIN_REQUIRE_PASSWORD_CHANGE_ON_BOOTSTRAP: "false"
       # Override .env.example placeholders — env vars take precedence over .env in pydantic-settings
-      JWT_SECRET_KEY: "ci-playwright-smoke-jwt-secret-not-for-production-use"
-      AUTH_ENCRYPTION_SECRET: "ci-playwright-smoke-auth-enc-secret-not-for-production"
+      JWT_SECRET_KEY: "ci-playwright-smoke-jwt-secret-not-for-production-use"  # pragma: allowlist secret
+      AUTH_ENCRYPTION_SECRET: "ci-playwright-smoke-auth-enc-secret-not-for-production"  # pragma: allowlist secret
 
     steps:
       - name: ⬇️ Checkout source

